internal/bminventory/inventory.go

```diff
@@ -1566,6 +1566,20 @@ func (b *bareMetalInventory) UpdateClusterInstallConfigInternal(ctx context.Cont
 		return nil, common.NewApiError(http.StatusBadRequest, err)
 	}
 
+	// Set install config overrides feature usage
+	usages, err := usage.Unmarshal(cluster.Cluster.FeatureUsage)
+	if err != nil {
+		log.WithError(err).Errorf("failed to read feature usage from cluster %s", params.ClusterID)
+	}
+
+	var installConfigOverrides map[string]interface{}
+	err = json.Unmarshal([]byte(params.InstallConfigParams), &installConfigOverrides)
+	if err != nil {
+		log.WithError(err).Errorf("failed to unmarshal install config overrides for feature usage")
+	} else if len(installConfigOverrides) > 0 {
+		b.setInstallConfigOverridesUsage(usages, installConfigOverrides, params.ClusterID)
+	}
```

Comment:
I think this could all be part of the same `setInstallConfigOverridesUsage` method.
Is there any reason to have it here?

Reply:
That's a good idea, moved it to the method!